assert/assertions_test.go

```diff
@@ -2786,6 +2795,36 @@ func TestEventuallyWithTTrue(t *testing.T) {
 	Len(t, mockT.errors, 0)
 }
 
+func TestEventuallyWithT_ConcurrencySafe(t *testing.T) {
+	mockT := new(errorsCapturingT)
+
+	condition := func(collect *CollectT) {
+		True(collect, false)
+	}
+
+	// To trigger race conditions, we run EventuallyWithT with a nanosecond tick.
+	False(t, EventuallyWithT(mockT, condition, 100*time.Millisecond, time.Nanosecond))
+	Len(t, mockT.errors, 2)
+}
+
+func TestEventuallyWithT_ReturnsTheLatestFinishedConditionErrors(t *testing.T) {
+	var calledOnce bool
+	condition := func(collect *CollectT) {
+		if calledOnce {
+			// Sleep to ensure that the second condition runs longer than timeout.
+			time.Sleep(time.Second)
+			return
+		}
```

Comment:
To better protect against race condition (the implementation of EventuallyWithT doesn't run overlapping conditions for now, but we don't want the test to be sensible to such a change), it would be better to avoid using a variable shared by multiple runs of the condition.

So it would be wise to use a channel of bools instead of `calledOnce`.

```go
mustSleep := make(chan bool, 2)
mustSleep <- false
mustSleep <- true
close(mustSleep)

...

if <-mustSleep {
    time.Sleep(time.Second)
}
```